src/backend/distributed/shared_library_init.c

```diff
@@ -2602,6 +2602,17 @@ OverridePostgresConfigAssignHooks(void)
 			OldApplicationNameAssignHook = stringVar->assign_hook;
 			stringVar->assign_hook = ApplicationNameAssignHook;
 		}
+
+		/*
+		 * Turn on GUC_REPORT for search_path. GUC_REPORT provides that an S (Parameter Status)
+		 * packet is appended after the C (Command Complete) packet sent from the server
+		 * for SET command. S packet contains the new value of the parameter
+		 * if its value has been changed.
+		 */
+		if (strcmp(var->name, "search_path") == 0)
```

Comment:
Not introduced by this PR, but this whole function can be simplified a lot. There's no need to loop over the full list of GUCs. We can simply call `find_option` to fetch the desired GUC from the hashmap instead. The primary storage is actually a hashmap and the `get_guc_variables` function above allocates a completely new array.

Reply:
Let's at least use `find_option` for `search_path`.  Then in a simple follow up PR we can do the refactor of the `application_name`.

Reply:
> Let's at least use `find_option` for `search_path` (outside of this loop). Then in a simple follow up PR we can do the refactor of the `application_name`.

It looks like find_option is not exported by postgres.

Reply:
Hmm you're right. It's only exported on the current master branch. So lets indeed keep the code as is. But lets add a comment that starting PG16 we can rewrite this using `find_option`.